src/WebJobs.Extensions.DurableTask/DurableTaskExtension.cs

```diff
@@ -89,15 +89,17 @@ public DurableTaskExtension()
         /// <param name="orchestrationServiceFactory">The factory used to create orchestration service based on the configured storage provider.</param>
         /// <param name="durableHttpMessageHandlerFactory">The HTTP message handler that handles HTTP requests and HTTP responses.</param>
         /// <param name="lifeCycleNotificationHelper">The lifecycle notification helper used for custom orchestration tracking.</param>
-        /// <param name="serializerSettingsFactory">The factory used to create JsonSerializerSettings.</param>
+        /// <param name="messageSerializerSettingsFactory">The factory used to create <see cref="JsonSerializerSettings"/> for message settings.</param>
+        /// /// <param name="errorSerializerSettingsFactory">The factory used to create <see cref="JsonSerializerSettings"/> for error settings.</param>
         public DurableTaskExtension(
             IOptions<DurableTaskOptions> options,
             ILoggerFactory loggerFactory,
             INameResolver nameResolver,
             IDurabilityProviderFactory orchestrationServiceFactory,
             IDurableHttpMessageHandlerFactory durableHttpMessageHandlerFactory = null,
             ILifeCycleNotificationHelper lifeCycleNotificationHelper = null,
-            ISerializerSettingsFactory serializerSettingsFactory = null)
+            IMessageSerializerSettingsFactory messageSerializerSettingsFactory = null,
+            IErrorSerializerSettingsFactory errorSerializerSettingsFactory = null)

```

Comment:
This works, but had you considered an alternate design where we have a single interface that defines two methods instead of having two interfaces that each define a single method?

Reply:
Yeah, I considered that. But if a customer is implementing the interface they'll need to implement both methods and I wanted to avoid that.

Reply:
We could use one interface assuming we did the following:

```csharp
public interface ISerializerSettingsFactory {
    JsonSerializerSettings GetMessageSerializerSettings();
    JsonSerializerSettings GetErrorSerializerSettings();
}
```

```csharp
public class DefaultSerializerSettingsFactory {
    public virtual JsonSerializerSettings GetMessageSerializerSettings() {
         // Default implementation here.
    }
    public virtual JsonSerializerSettings GetErrorSerializerSettings() {
         // Default implementation here.
    }
}
```

Then customers could just make a subclass of DefaultSerializerSettingsFactory, and only override the method they want to.

Reply:
Good idea, I hadn't thought of that option!

I'm going to leave the current implementation for consistency reasons though, unless anyone objects.